src/ccm.rs

```diff
@@ -204,17 +210,18 @@ impl CcmMode {
             ccm_cbc_mac(&mut b, associated_data, true, &self.cipher);
         }
         if plen > 0 {
-            ccm_cbc_mac(&mut b, &out[..plen - self.mlen], false, &self.cipher);
+            ccm_cbc_mac(&mut b, payload, false, &self.cipher);
         }
 
         // Comparing the received tag and the computed one
-        if b[..self.mlen] != tag[..self.mlen] {
+        use subtle::ConstantTimeEq;
+        if b[..tlen].ct_eq(&t[..tlen]).unwrap_u8() == 0 {
```

Comment:
Note: made the MAC comparison here constant time using the [subtle](https://docs.rs/subtle/) crate

Reply:
Very nice. Quick stylistic question here, because I never read anything about it: is there a consensus on when to import something like the `ConstantTimeEq` trait directly before the code using it, as opposed to the top of the file? When it's only used once?

Reply:
For the `subtle::ConstantTimeEq` trait specifically, I've usually seen it imported in this style, I assume with the intent of making it easier for a reviewer to see what's happening.

Reply:
Ah, that makes sense. Thanks!